homeassistant/components/yeelight/__init__.py

```diff
@@ -196,6 +198,26 @@ def _async_load_platforms():
         device = await _async_get_device(hass, host, entry)
     entry_data[DATA_DEVICE] = device
 
+    # start listening for local pushes
+    try:
+        await device.bulb.async_listen(device.update_callback)
```

Comment:
Can you cover this line by patching `async_listen` to raise `BlubException`

Reply:
I tried, but I can't get the tests to work, I tried this:
```
async def test_async_listen_error(hass: HomeAssistant):
    """Test the async listen error."""
    config_entry = MockConfigEntry(domain=DOMAIN, data=CONFIG_ENTRY_DATA)
    config_entry.add_to_hass(hass)

    mocked_bulb = _mocked_bulb()
    mocked_bulb.async_listen = MagicMock(return_value=Future())
    mocked_bulb.async_listen.return_value.set_exception(BulbException)
    
    with _patch_discovery(MODULE), patch(
        f"{MODULE}.AsyncBulb", return_value=mocked_bulb
    ), pytest.raises(ConfigEntryNotReady):
        await hass.config_entries.async_setup(config_entry.entry_id)
```

FAILED tests/components/yeelight/test_init.py::test_async_listen_error[pyloop] - Failed: DID NOT RAISE <class 'homeassistant.exceptions.ConfigEntryNotReady'>
FAILED tests/components/yeelight/test_init.py::test_async_listen_error[pyloop] - homeassistant.exceptions.ConfigEntryNotReady

Stupidly enough it seems like it is complaining that the test did not raise and raised an exception.

Reply:
Assert the config entry is in retry state instead.  The `ConfigEntryNotReady` exception will be consumed by the config entry code itself.

Reply:
`assert entry.state is ConfigEntryState.SETUP_RETRY`

Reply:
I changed it to:
```
async def test_async_listen_error(hass: HomeAssistant):
    """Test the async listen error."""
    config_entry = MockConfigEntry(domain=DOMAIN, data=CONFIG_ENTRY_DATA)
    config_entry.add_to_hass(hass)

    mocked_bulb = _mocked_bulb()
    mocked_bulb.async_listen = MagicMock(return_value=Future())
    mocked_bulb.async_listen.return_value.set_exception(BulbException)
    
    with _patch_discovery(MODULE), patch(
        f"{MODULE}.AsyncBulb", return_value=mocked_bulb
    ):
        await hass.config_entries.async_setup(config_entry.entry_id)

    assert config_entry.state is ConfigEntryState.SETUP_RETRY
```
but still get these errors:

FAILED tests/components/yeelight/test_init.py::test_async_listen_error[pyloop] - AssertionError: assert <ConfigEntryState.LOADED: 'loaded'> is <ConfigEntryState.SETUP_RETRY: 'setup_retry'>
FAILED tests/components/yeelight/test_init.py::test_async_listen_error[pyloop] - homeassistant.exceptions.ConfigEntryNotReady